filebeat/input/filestream/input_test.go

```diff
@@ -0,0 +1,175 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package filestream
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	loginp "github.com/elastic/beats/v7/filebeat/input/filestream/internal/input-logfile"
+	v2 "github.com/elastic/beats/v7/filebeat/input/v2"
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/beats/v7/libbeat/statestore"
+	"github.com/elastic/beats/v7/libbeat/statestore/storetest"
+	conf "github.com/elastic/elastic-agent-libs/config"
+	"github.com/elastic/elastic-agent-libs/logp"
+)
+
+// runFilestreamBenchmark runs the entire filestream input with the in-memory registry and the test pipeline.
+// `testID` must be unique for each test run
+// `cfg` must be a valid YAML string containing valid filestream configuration
+// `expEventCount` is an expected amount of produced events
+func runFilestreamBenchmark(t testing.TB, testID string, cfg string, expEventCount int) {
+	logger := logp.L()
+	c, err := conf.NewConfigWithYAML([]byte(cfg), cfg)
+	require.NoError(t, err)
+
+	p := Plugin(logger, createTestStore(t))
+	input, err := p.Manager.Create(c)
+	require.NoError(t, err)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	context := v2.Context{
+		Logger:      logger,
+		ID:          testID,
+		Cancelation: ctx,
+	}
+
+	connector, eventsDone := newTestPipeline(expEventCount)
+	done := make(chan struct{})
+	go func() {
+		err := input.Run(context, connector)
```

Comment:
nit.  Right now the benchmark is capturing the time spent with the config, setting up manager etc.  Might be worth using [StopTimer](https://cs.opensource.google/go/go/+/go1.21.5:src/testing/benchmark.go;l=135) and StartTimer to limit the reporting the time spent in `input.Run`.  I'm sure the setup time is roughly constant so not a huge deal, but it might make it easier to see small changes